### ISSUE TITLE
")" unexpectedly indented in expression but not in `if`

### ISSUE THREAD

**Issue by mathjeff** [pinterest/ktlint#920]
Given ktlint 0.39.0 and some code that looks like this:

```
val pass = if (condition1) {                 
        (
            condition2
        )
    } else {
        condition3
    }
```

## Expected Behavior:
this code seems to be formatted reasonably to us; we think ktlint should report no problems.

## Observed Behavior:
Instead we're observing it to request indenting "condition2" and ")" to get this:

```
val pass = if (condition1) {                 
        (
                condition2
            )
    } else {
        condition3
    }
```

At first I thought this seemed reasonable to me because it was presumably being treated as a line continuation, but then I checked what would happen if condition1 were multiple lines, and found in that case that ktlint requested that the parentheses of the `if` statement should be lined up:

```
if (
    condition1
) { // indendation matching the corresponding "("
    (
            condition2
        ) // indentation not matching the corresponding "("
} else {
    condition3
}
```

## Your Environment
We encounter this when upgrading from ktlint 0.36.0 to 0.39.0

The full code in question: https://android-review.googlesource.com/c/platform/frameworks/support/+/1432017/2/compose/animation/animation-core/src/test/java/androidx/compose/animation/core/SpringEstimationTest.kt#139

**Comment by romtsn** [comment]
That is a known quirk of the Kotlin ide plugin formatter (whose behavior we are trying to match), please check #810 for details. I've also filed an issue on [youtrack](https://youtrack.jetbrains.com/issue/KT-41316) and have to close this until further notice from JB, as it seems to be expected behavior.

**Comment by oakkitten** [comment]
I gave that YouTrack issue a thumbs up, which brought the total number of thumbs up to two. It's been two years, perhaps it's worth considering whether it could be okay to allow deviating from the IDE formatter in cases such as this? This looks super weird 🤪

**Comment by mklnwt** [comment]
I also gave it a thumbs up. Any chance we could at least add a flag to allow deviation?

**Comment by paul-dingemans** [comment]
Reopened issue for investigation. If it gets solved, it will only be solved for the new code style `ktlint_official`.
